cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -398,29 +401,46 @@ var (
 )
 
 func isCoreDNSVersionSupported(client clientset.Interface) (bool, error) {
-	isValidVersion := true
-	coreDNSPodList, err := client.CoreV1().Pods(metav1.NamespaceSystem).List(
-		context.TODO(),
-		metav1.ListOptions{
-			LabelSelector: "k8s-app=kube-dns",
-		},
-	)
-	if err != nil {
-		return false, errors.Wrap(err, "unable to list CoreDNS pods")
-	}
+	passed := true
+	var lastError error
+	err := wait.PollImmediate(constants.APICallRetryInterval, 10 * time.Second, func() (bool, error) {
+		coreDNSPodList, err := client.CoreV1().Pods(metav1.NamespaceSystem).List(
```

Comment:
i think we can have the call to get the list outside of the poll.
the purpose of the poll would be to get the pods in a Running state and have containers where we can get the ImageIDs.

Reply:
IMO, the poll function should be used to obtain a list of pods, and whether the list is empty as a loop condition.

Reply:
ok, agreed.